### PR TITLE
smb3-foundry: init at 1.2

### PR DESCRIPTION
**Pull request by TomaSajt**
###### Description of changes

This PR adds the smb3-foundry package and updates the py65 package.

Homepage for smb3-foundry: https://github.com/mchlnix/SMB3-Foundry

---

The py65 package was originally only declared as a pythonApplication, now, it has been added to pythonPackages, and then also converted into a pythonApplication.

The py65 version was not changed, however I added some testing and used fetchFromGitHub instead.

The smb3-foundry package needs an SMB3 rom, so if you want to test it out, you'll have to get one.

---
I made an extra commit for py65 which only moves the file, so that git can recognize the original file was just moved. This was done to preserve the git history for the file. (hopefully)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by TomaSajt** [comment]
Closes https://github.com/NixOS/nixpkgs/issues/241007

File: pkgs/applications/misc/smb3-foundry/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, python3
+, makeWrapper
+}:
+let
+  pythonEnv = (python3.withPackages (ps: with ps;[
```
Comment (AndersonTorres) [inline pkgs/applications/misc/smb3-foundry/default.nix:8]:
```suggestion
  pythonEnv = (python3.withPackages (ps: with ps; [
```

File: pkgs/applications/misc/smb3-foundry/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    repo = "SMB3-Foundry";
+    rev = "refs/tags/${version}";
+    hash = "sha256-iqqIyGp/sqWgShxk52omVcn7Q3WL2hK8sTLH4dashLE=";
+  };
+
+  patches = [ ./fix-relative-dirs.patch ];
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
```
Comment (AndersonTorres) [inline pkgs/applications/misc/smb3-foundry/default.nix:29]: Don't forget the runHooks.
```suggestion
  installPhase = ''
    runHook preInstall

```

File: pkgs/applications/misc/smb3-foundry/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    mkdir -p $out/app
+    cp -R smb3parse foundry scribe data doc smb3-foundry.py smb3-scribe.py $out/app
+    echo "${version}" > $out/app/VERSION
+
+    makeWrapper ${pythonEnv}/bin/python $out/bin/smb3-foundry \
+          --add-flags "$out/app/smb3-foundry.py"
+    makeWrapper ${pythonEnv}/bin/python $out/bin/smb3-scribe \
+          --add-flags "$out/app/smb3-scribe.py"
+  '';
```
Comment (AndersonTorres) [inline pkgs/applications/misc/smb3-foundry/default.nix:38]:
```suggestion

    runHook postInstall
  '';
```

File: pkgs/applications/misc/smb3-foundry/default.nix [hunk 4]
```diff
@@ -0,0 +1,48 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, python3
+, makeWrapper
+}:
+let
```
Comment (AndersonTorres) [inline pkgs/applications/misc/smb3-foundry/default.nix:7]:
```suggestion

let
```

File: pkgs/applications/misc/smb3-foundry/default.nix [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  };
+
+  patches = [ ./fix-relative-dirs.patch ];
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
+    mkdir -p $out/app
+    cp -R smb3parse foundry scribe data doc smb3-foundry.py smb3-scribe.py $out/app
+    echo "${version}" > $out/app/VERSION
```
Comment (AndersonTorres) [inline pkgs/applications/misc/smb3-foundry/default.nix:32]: Is this needed?

File: pkgs/applications/misc/smb3-foundry/default.nix [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+, makeWrapper
+}:
+let
+  pythonEnv = (python3.withPackages (ps: with ps;[
+    pyside6
+    py65
+    qdarkstyle
+  ]));
+in
+stdenv.mkDerivation rec {
```
Comment (AndersonTorres) [inline pkgs/applications/misc/smb3-foundry/default.nix:14]: I recommend the rec-less version:


```suggestion
stdenv.mkDerivation (finalAttrs: {
```

File: pkgs/applications/misc/smb3-foundry/default.nix [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  pname = "smb3-foundry";
+  version = "1.2";
+
+  src = fetchFromGitHub {
+    owner = "mchlnix";
+    repo = "SMB3-Foundry";
+    rev = "refs/tags/${version}";
```
Comment (AndersonTorres) [inline pkgs/applications/misc/smb3-foundry/default.nix:21]:
```suggestion
    rev = "refs/tags/${finalAttrs.version}";
```

File: pkgs/applications/misc/smb3-foundry/default.nix [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    cp -R smb3parse foundry scribe data doc smb3-foundry.py smb3-scribe.py $out/app
+    echo "${version}" > $out/app/VERSION
+
+    makeWrapper ${pythonEnv}/bin/python $out/bin/smb3-foundry \
+          --add-flags "$out/app/smb3-foundry.py"
+    makeWrapper ${pythonEnv}/bin/python $out/bin/smb3-scribe \
+          --add-flags "$out/app/smb3-scribe.py"
+  '';
+
+  meta = with lib; {
```
Comment (AndersonTorres) [inline pkgs/applications/misc/smb3-foundry/default.nix:40]:
```suggestion
  meta = {
```

File: pkgs/applications/misc/smb3-foundry/default.nix [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    homepage = "https://github.com/mchlnix/SMB3-Foundry";
+    description = "A modern Super Mario Bros. 3 Level Editor";
+    changelog = "https://github.com/mchlnix/SMB3-Foundry/releases/tag/${version}";
```
Comment (AndersonTorres) [inline pkgs/applications/misc/smb3-foundry/default.nix:43]:
```suggestion
    changelog = "https://github.com/mchlnix/SMB3-Foundry/releases/tag/${finalAttrs.version}";
```

File: pkgs/applications/misc/smb3-foundry/default.nix [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    license = licenses.gpl3Only;
+    platforms = platforms.unix;
+    maintainers = with maintainers; [ tomasajt ];
```
Comment (AndersonTorres) [inline pkgs/applications/misc/smb3-foundry/default.nix:46]:
```suggestion
    license = lib.licenses.gpl3Only;
    platforms = lib.platforms.unix;
    maintainers = with lib.maintainers; [ tomasajt ];
```

File: pkgs/applications/misc/smb3-foundry/fix-relative-dirs.patch [hunk 11]
```diff
@@ -0,0 +1,34 @@
+diff --git a/foundry/gui/WarningList.py b/foundry/gui/WarningList.py
```
Comment (AndersonTorres) [inline pkgs/applications/misc/smb3-foundry/fix-relative-dirs.patch:1]: Can this patch be upstreamed?

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/smb3-foundry/fix-relative-dirs.patch (hunk 11 above)
Comment (TomaSajt) [reply pkgs/applications/misc/smb3-foundry/fix-relative-dirs.patch:1]: It could be, I'll open a PR

Comment (TomaSajt) [reply pkgs/applications/misc/smb3-foundry/fix-relative-dirs.patch:1]: Okay, I see that the repo is quite active, so I'll wait out the PR being merged, and then use fetchpatch

File: pkgs/development/python-modules/py65/default.nix [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+    hash = "sha256-WLs3TAZovuphWZIvMvM3CZnqg1aZfMF4Yrqw46k+bLA=";
+  };
+
+  postPatch = ''
+    substituteInPlace py65/tests/test_monitor.py \
+          --replace "test_argv_rom" "dont_test_argv_rom" \
+          --replace "test_argv_combination_rom_mpu" "dont_test_argv_combination_rom_mpu"
+  '';
+
+  meta = with lib; {
```
Comment (AndersonTorres) [inline pkgs/development/python-modules/py65/default.nix:27]:
```suggestion
  meta = {
```

File: pkgs/development/python-modules/py65/default.nix [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+    homepage = "https://py65.readthedocs.io/";
+    description = "Emulate 6502-based microcomputer systems in Python";
+    longDescription = ''
+      Py65 includes a program called Py65Mon that functions as a machine
+      language monitor. This kind of program is sometimes also called a
+      debugger. Py65Mon provides a command line with many convenient commands
+      for interacting with the simulated 6502-based system.
+    '';
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ AndersonTorres tomasajt ];
```
Comment (AndersonTorres) [inline pkgs/development/python-modules/py65/default.nix:37]:
```suggestion
    license = lib.licenses.bsd3;
    maintainers = with lib.maintainers; [ AndersonTorres tomasajt ];
```

File: maintainers/maintainer-list.nix [hunk 14]
```diff
@@ -16638,6 +16638,14 @@
     githubId = 8577941;
     name = "Kevin Rauscher";
   };
+  tomasajt = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:16641]: No e-mail?

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 14 above)
Comment (TomaSajt) [reply maintainers/maintainer-list.nix:16641]: I don't currently have a business email address, when I do, I'll add it

**Review by AndersonTorres — APPROVED** [review]

**Pull request merged**